hack/testmanifestfmt/main.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/hc-install/product"
+	"github.com/hashicorp/hc-install/releases"
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+func init() {
+	flag.Parse()
+}
+
+func main() {
+	ctx := context.Background()
+
+	log.Println("[INFO] installing terraform")
+
+	installer := &releases.ExactVersion{
+		Product: product.Terraform,
+		Version: version.Must(version.NewVersion("1.0.6")),
+	}
+
+	execPath, err := installer.Install(ctx)
+	if err != nil {
+		log.Fatalf("[ERROR] unable to intsall terraform: %s", err)
+	}
+
+	tmpdir := os.TempDir()
+	defer func() { _ = os.RemoveAll(tmpdir) }()
+
+	tf, err := tfexec.NewTerraform(tmpdir, execPath)
+	if err != nil {
+		log.Fatalf("[ERROR] unable to create terraform handle: %s", err)
+	}
+
+	log.Println("[INFO] installed terraform, formatting files")
+
+	for i := range flag.Args() {
+		path := flag.Args()[i]
+
+		log.Println("[INFO] formatting ", path)
+		if err = reformatFileAtPath(ctx, path, tf); err != nil {
+			log.Println("[ERROR]: unable to format ", path, ": ", err)
+		}
+	}
+
+	log.Println("[INFO] formatted all files")
+}
+
+func reformatFileAtPath(ctx context.Context, path string, tf *tfexec.Terraform) error {
+	file, err := os.OpenFile(path, os.O_RDWR, 0755)
+	if err != nil {
+		return fmt.Errorf("unable to open file at path '%s': %w", path, err)
+	}
+	defer func() { _ = file.Close() }()
+
+	fset := token.NewFileSet()
+	parsed, err := parser.ParseFile(fset, "src.go", file, parser.ParseComments)
+	if err != nil {
+		return fmt.Errorf("unable to parse file at path '%s': %w", path, err)
+	}
+
+	var hadFmtErr bool
+	ast.Inspect(parsed, func(n ast.Node) bool {
+		lit, ok := n.(*ast.BasicLit)
+		if !ok {
+			return true
+		}
+
+		if lit.Kind != token.STRING {
+			return false
+		}
+		// TODO: better way to check if the string is a terraform manifest
+		if !isTerraformManifestMaybe(lit.Value) {
+			return false
+		}
+
+		unquoted := strings.Trim(lit.Value, "`")
+
+		if formatted, err := tf.FormatString(ctx, unquoted); err != nil {
+			log.Println("[ERROR]: unable to format node ", unquoted, ": ", err)
+			hadFmtErr = true
+		} else {
+			lit.Value = "`" + formatted + "`"
+		}
+		return false
+	})
+
+	if hadFmtErr {
+		return fmt.Errorf("unable to format file at path '%s'", path)
+	}
+
+	if err := file.Truncate(0); err != nil {
+		return fmt.Errorf("unable to truncate file at path '%s': %w", path, err)
+	}
+	if _, err := file.Seek(0, 0); err != nil {
+		return fmt.Errorf("unable to seek beginning of file at path '%s': %w", path, err)
+	}
+
+	format.Node(file, fset, parsed)
+
+	return nil
+}
+
+func isTerraformManifestMaybe(s string) bool {
```

Comment:
I'd argue that `Maybe` is redundant here. `isTerraformManifest` clearly explains what the function is about.

Reply:
done